### PR TITLE
feat(check:policy): Prevent .js file extension

### PR DESCRIPTION
**Pull request by tylerbutler**
Adds a policy handler that checks for JavaScript source files that just use the .js file extension. Such files may be interpreted by node as either CommonJS or ESM based on the `type` field in the nearest package.json file. However, this is can lead to confusing behavior when updating packages to use `type`. Using explicit file extensions reduces ambiguity and ensures a CJS file isn't suddenly treated like an ESM file.

### CODE REVIEW

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/index.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import { handlers as copyrightFileHeaderHandlers } from "./copyrightFileHeader";
 import { handler as dockerfilePackageHandler } from "./dockerfilePackages";
 import { handler as fluidCaseHandler } from "./fluidCase";
 import { handlers as lockfilesHandlers } from "./lockfiles";
+import { handler as noJsFileHandler } from "./noJsFiles";
```
Comment (jason-ha) [inline build-tools/packages/build-tools/src/repoPolicyCheck/handlers/index.ts:11]: We should always be using full paths now (okay if some aren't in my opinion until package is wholly updated; just good practice) Okay not to change.

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Handler } from "../common";
+
+/**
+ * A policy handler that checks for JavaScript source files that just use the .js file extension. Such files may be
+ * interpreted by node as either CommonJS or ESM based on the `type` field in the nearest package.json file. However
+ * this is not deterministic. Using file extensions reduces ambiguity and makes the behavior deterministic.
```
Comment (jason-ha) [inline build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts:10]: What is not deterministic? You just mean that _we_ might change the package type?

**Review by jason-ha — APPROVED** [review]
I like it. :)

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/index.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/repoPolicyCheck/handlers/index.ts:11]: Prefer to fix these sorts of things all at once so someone else doesn't see it half one way and half another and get confused as to why they're inconsistent.

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts (hunk 2 above)
Comment (jason-ha) [reply build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts:10]: Doesn't seem not related. That is the system.
Brings to mind the missing fluid build dependency on package.json. Change package.json as you like. Build cache doesn't care.

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ */
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts:11]:
```suggestion
 * A policy handler that checks for JavaScript source files that just use the .js file extension. Such files may be
 * interpreted by node as either CommonJS or ESM based on the `type` field in the nearest package.json file. This
 * can create unexpected behavior for JS files; changing the package.json nearest to one will change how the JS
 * is processed by node. Using explicit file extensions reduces ambiguity and ensures a CJS file isn't suddenly treated
 * like an ESM file..
 */
```

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * Licensed under the MIT License.
+ */
+import { Handler } from "../common";
+
+/**
+ * A policy handler that checks for JavaScript source files that just use the .js file extension. Such files may be
+ * interpreted by node as either CommonJS or ESM based on the `type` field in the nearest package.json file. This
+ * can create unexpected behavior for JS files; changing the package.json nearest to one will change how the JS
+ * is processed by node. Using explicit file extensions reduces ambiguity and ensures a CJS file isn't suddenly treated
+ * like an ESM file..
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts:12]:
```suggestion
 * like an ESM file.
```

File: build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts (hunk 2 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/repoPolicyCheck/handlers/noJsFiles.ts]: Clarified the comment. I understand that this is node's behavior, but it is not obvious, and it's also new(ish).

**Pull request merged**